File: frontend/lib/models/todo.dart
class Todo {
  final int id;
  String title;
  bool isCompleted;
  bool isDeleted;

  Todo({
    required this.id,
    required this.title,
    required this.isCompleted,
    required this.isDeleted,
  });

  factory Todo.fromJson(Map<String, dynamic> json) {
    return Todo(
      id: json['id'],
      title: json['title'],
      isCompleted: json['isCompleted'] ?? false,
      isDeleted: json['isDeleted'] ?? false,
    );
  }
}

File: Backend/models/Todo.js
// models/Todo.js
const mongoose = require('mongoose');

const todoSchema = new mongoose.Schema({
  title: { type: String, required: true },
  isCompleted: { type: Boolean, default: false },
  isDeleted: { type: Boolean, default: false },
  createdAt: { type: Date, default: Date.now }
});

module.exports = mongoose.model('Todo', todoSchema);
